pkg/urbit/vere/io/ames.c

```diff
@@ -652,23 +690,42 @@ _ames_lane_scry_cb(void* vod_p, u3_noun nun)
   u3_panc* pac_u = vod_p;
   u3_weak  las = u3r_at(7, nun);
 
+  pac_u->sam_u->foq_d--;
+
   //  if scry fails, remember we can't scry, and just inject the packet
   //
   if (u3_none == las) {
-    pac_u->sam_u->foq_d--;
-    u3l_log("ames: giving up scry\n");
-    pac_u->sam_u->see_o = c3n;
+    if (5 < ++pac_u->sam_u->saw_d) {
+      u3l_log("ames: giving up scry\n");
+      pac_u->sam_u->see_o = c3n;
+    }
     _ames_put_packet(pac_u->sam_u,
                      _ames_serialize_packet(pac_u, c3n),
                      pac_u->ore_u);
     _ames_panc_free(pac_u);
   }
-  //  if there is a lane, forward the packet on it
-  //
-  else if (u3_nul != las) {
-    _ames_forward(pac_u, u3k(las));
+  else {
+    if (0 < pac_u->sam_u->saw_d) {
+      pac_u->sam_u->saw_d--;
+    }
```

Comment:
Do we want any successful scry to just reset this counter to 0? I had thought this feature was mostly to detect ames namespace mismatches, are you picturing it more as general purpose backoff?